### PR TITLE
Additional source and playlist collection fix

### PR DESCRIPTION
**Pull request by Mrlaminat**

### CODE REVIEW

File: src/YoutubeDl.php [hunk 1]
```diff
@@ -116,9 +116,12 @@ public function download(Options $options): VideoCollection
         $process->run(function (string $type, string $buffer) use (&$currentVideo, &$parsedData, &$progressTarget): void {
             ($this->debug)($type, $buffer);
 
-            if (preg_match('/\[(.+)]\s(.+):\sDownloading webpage/', $buffer, $match) === 1) {
+            if (preg_match('/\[(.+)]\s(.+):\sDownloading webpage/', $buffer, $match) === 1
+                || preg_match('/\[(.+)]\s(.+):\sDownloading pc webpage/', $buffer, $match) === 1
+            ) {
```
Comment (norkunas) [inline src/YoutubeDl.php:121]:
```suggestion
            if (preg_match('/\[(?<extractor>.+)]\s(?<id>.+):\sDownloading( pc )?webpage/', $buffer, $match) === 1) {
```

File: src/YoutubeDl.php [hunk 2]
```diff
@@ -116,9 +116,12 @@ public function download(Options $options): VideoCollection
...
                 if ($currentVideo !== null) {
                     $parsedData[] = $currentVideo;
+                    $currentVideo = null;
                 }
 
                 $currentVideo['extractor'] = $match[1];
```
Comment (norkunas) [inline src/YoutubeDl.php:127]:
```suggestion
                $currentVideo['extractor'] = $match['extractor'];
                $currentVideo['id'] = $match['id'];
```

Comment (norkunas) [reply src/YoutubeDl.php:127]: can't suggest for two lines, so just `$currentVideo['id'] = $match[2];` with `$currentVideo['id'] = $match['id'];`

**Review by norkunas — CHANGES_REQUESTED** [review]
Thanks for your findings :)

**Comment by norkunas** [comment]
Would be nice to add a test that it properly parses output with the `pc` in it

File: src/YoutubeDl.php (hunk 1 above)
Comment (Mrlaminat) [reply src/YoutubeDl.php:121]: Yeah, this one looks much better :)

**Comment by Mrlaminat** [comment]
> Would be nice to add a test that it properly parses output with the `pc` in it

Need to check how tests are implemented for your system. Is it has some prepared URL?

**Comment by norkunas** [comment]
> > Would be nice to add a test that it properly parses output with the `pc` in it
> 
> Need to check how tests are implemented for your system. Is it has some prepared URL?

No, it does not do real requests, only tests how it works with the stream output

**Comment by Mrlaminat** [comment]
@norkunas Fixed regex. 

PS. I would recommend you add a regex container, from which all regexes could be tested. It will help to cover more supported platforms without huge regex complexity. If you worrying about performance, maybe you can add a function flag ("useMultiplePlatformRegex", by default false).

**Comment by norkunas** [comment]
@Mrlaminat I'm not against it, but currently regexes are simple enough not to introduce your suggested regex container. We'll see in future if it become hard to maintain then it probably will be time for that.
Will you be able to write a test for this?

**Comment by norkunas** [comment]
Also I'm curious which source adds the `pc` to that sentence? Or is it with the yt-dlp that it adds different source to it?

**Comment by Mrlaminat** [comment]
> @Mrlaminat I'm not against it, but currently regexes are simple enough not to introduce your suggested regex container. We'll see in future if it become hard to maintain then it probably will be time for that. Will you be able to write a test for this?

It was just suggestion for the future, right now is good enough!

I just fixed code to pass already existed tests, from my local env I even can't run them with xDebug so it's difficult to find how to create a new one.

LIke for me it works perfect, and it's not broke usuall youtube flow, so looks like we are okay :)

**Comment by norkunas** [comment]
just run with XDEBUG_MODE=off :)

**Comment by Mrlaminat** [comment]
But I need to enable, not disable :) 
Anyway, what do you want from new tests? why old one is not satisfied you ?

**Comment by norkunas** [comment]
Oh, and why do you need to run with it ? :D
Because there are not tests that covers the part with `Downloading pc webpage` so it's better to have at least 1 test to not break this accidentally in future :)

**Comment by Mrlaminat** [comment]
> Oh, and why do you need to run with it ? :D Because there are not tests that covers the part with `Downloading pc webpage` so it's better to have at least 1 test to not break this accidentally in future :)

I'm not good at unit testing, finding why the previose failing was paitful :)

**Comment by norkunas** [comment]
Ok then if you'd like for me to add the test, could you run the lib in debug mode and give me all output it prints?

**Comment by Mrlaminat** [comment]
Whatever, Good time for learning. I'll add it

**Comment by norkunas** [comment]
Why you have closed this?

**Comment by Mrlaminat** [comment]
@norkunas changed my mind. Sorry for the confusion and thank you

**Comment by norkunas** [comment]
Just not to bother you, open and I'll merge as is :bow:

**Review by norkunas — APPROVED** [review]

**Comment by norkunas** [comment]
Thank you!

**Comment by Mrlaminat** [comment]
Man, that's a Great project, thank you for developing and maintaining it!

**Pull request merged**